packages/flutter_tools/lib/src/protocol_discovery.dart

```diff
@@ -84,19 +74,35 @@ class ProtocolDiscovery {
     if (index >= 0) {
       try {
         uri = Uri.parse(line.substring(index + prefix.length));
-      } catch (_) {
-        // Ignore errors.
+      } catch (error) {
+        _stopScrapingLogs();
```

Comment:
My thought was that if we ever see "Observatory listening on http://" and can't parse the associated URI, it's most likely because something went wrong, and we should throw.